plugins/event_source/cmd.py

```diff
@@ -0,0 +1,92 @@
+"""
+cmd.py
+
+An ansible-rulebook event source plugin for running a custom command.
+It can send events to the rule engine from the stdout.
+
+Arguments:
+    command: str
+        The string with the command to execute
+    send_output: bool, default: false
+        Send events from the stdout.
+        Each event must be a serialized JSON string
+        ended with a newline character ('\n')
+
+Examples:
+---
+- name: Example with output proccessing
+  hosts: all
+  sources:
+    - ansible.eda.cmd:
+        command: node my_custom_plugin.js
+        send_output: true
+  rules:
+    - name: r1
+      condition: event.cmd.status == "Running"
+      action:
+        print_event:
+
+
+---
+- name: Example combined with webhook plugin
+  hosts: all
+  sources:
+    - ansible.eda.webhook:
+        port: 6080
+    - ansible.eda.cmd:
+        command: node my_custom_plugin.js --url http://localhost:6080/eda
+  rules:
+    - name: r1
+      condition: event.payload.status == "Running"
+      action:
+        print_event:
+"""
+
+import asyncio
+import json
+import logging
+from typing import Any, Dict
+
+LOGGER = logging.getLogger("ansible.eda.cmd")
+
+
+async def process_output(
+    proc: asyncio.subprocess.Process, queue: asyncio.Queue
+):
+    # proc.stdout can be None, this is not our case
+    output = await proc.stdout.readline()
+    if output:
+        try:
+            event = json.loads(output.decode())
+            await queue.put({"cmd": event})
+        except json.JSONDecodeError as e:
+            LOGGER.error("Can not deserialize JSON data: %s", e)
+
+
+async def main(queue: asyncio.Queue, args: Dict[str, Any]):
+    command = str(args["command"])
+    send_output = False
+
+    if "send_output" in args:
+        send_output = args["send_output"]
+
+    proc = await asyncio.subprocess.create_subprocess_shell(
+        command, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+    )
+
+    LOGGER.info("Process started")
+
+    while proc.returncode is None:
```

Comment:
You can use `proc.wait()` to wait for a process to finish, instead of running a loop with sleep.

Reply:
The loop is necessary because I need to read the stdout while the process is running. The goal is receive each event ([one line](https://github.com/ansible/event-driven-ansible/pull/65/files#diff-0785115d178716a59957832e490c7fc065856df1abe0cc4caa1a117869a64fa2R57)) at a time during the lifetime of the process.